app/controllers/manageBusinesses/add/IncomeSourceAddedController.scala

```diff
@@ -120,27 +123,57 @@ class IncomeSourceAddedController @Inject()(val authActions: AuthActions,
     }
   }
 
-  private def handleSuccess(incomeSourceId: IncomeSourceId, incomeSourceType: IncomeSourceType, businessName: Option[String],
-                            showPreviousTaxYears: Boolean, isAgent: Boolean, sessionData: UIJourneySessionData, reportingMethod: ChosenReportingMethod
+  private def handleSuccess(
+                             incomeSourceId: IncomeSourceId,
+                             incomeSourceType: IncomeSourceType,
+                             businessName: Option[String],
+                             showPreviousTaxYears: Boolean,
+                             isAgent: Boolean,
+                             sessionData: UIJourneySessionData,
+                             reportingMethod: ChosenReportingMethod
                            )(implicit user: MtdItUser[_], errorHandler: ShowInternalServerError): Future[Result] = {
-    val oldAddIncomeSourceSessionData = sessionData.addIncomeSourceData.getOrElse(AddIncomeSourceData())
-    val updatedAddIncomeSourceSessionData = oldAddIncomeSourceSessionData.copy(journeyIsComplete = Some(true))
+
+
+    val oldAddIncomeSourceSessionData: AddIncomeSourceData = sessionData.addIncomeSourceData.getOrElse(AddIncomeSourceData())
+    val updatedAddIncomeSourceSessionData: AddIncomeSourceData = oldAddIncomeSourceSessionData.copy(journeyIsComplete = Some(true))
     val uiJourneySessionData: UIJourneySessionData = sessionData.copy(addIncomeSourceData = Some(updatedAddIncomeSourceSessionData))
-    sessionService.setMongoData(uiJourneySessionData).flatMap { _ =>
-      uiJourneySessionData.addIncomeSourceData.flatMap(_.dateStarted) match {
+
+        println("*********   " + oldAddIncomeSourceSessionData)
+        println("=========  " + updatedAddIncomeSourceSessionData)
+        println("+++++++++   " + uiJourneySessionData)
+        println("@@@@@@@@@   " + user.incomeSources.businesses.map(_.latencyDetails))
+
+    val incomeSourceBeingAddedLatencyDetails: Option[LatencyDetails] =
+      user.incomeSources.businesses.flatMap(_.latencyDetails).headOption
+
+    val latencyCombinationContentHelper: IncomeSourceLatencyStatus =
+      incomeSourceBeingAddedLatencyDetails match {
+        case Some(LatencyDetails(_, _, "Q", _, "Q")) => QuarterlyQuarterly
+        case Some(LatencyDetails(_, _, "A", _, "Q")) => AnnualQuarterly
+        case Some(LatencyDetails(_, _, "Q", _, "A")) => QuarterlyAnnual
+        case Some(LatencyDetails(_, _, "A", _, "A")) => AnnualAnnual
+        case _ => UnknownLatency
+      }
```

Comment:
we can maybe try to define a way to determine the scenario, and show content based on the latency combination. Or does it have to be from session?